Content/Archaeology/Artifact.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Core;
+using StarlightRiver.Helpers;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.GameInput;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+using System;
+
+namespace StarlightRiver.Content.Archaeology
+{
+    public abstract class Artifact : ModTileEntity
+	{
+        public bool displayedOnMap = false; //Whether or not the artifact is displayed on the map
+
+        public virtual bool CanBeRevealed() => true; //Whether or not the artifact can be revealed by the archaeologist's map. Set to false if the artifact has a special reveal condition
+
+        public virtual string TexturePath => AssetDirectory.Archaeology + Name;
+
+        public virtual string MapTexturePath => AssetDirectory.Archaeology + "DigMarker"; //Texture path the artifact uses on the map when revealed
+
+        public virtual Vector2 Size { get; } //Size of the artifact. In world coordinates, not tile coordinates
```

Comment:
World coordiantes are generally stored in Point16 variables -- should we be using the same here?

Reply:
By world coordinates I meant the position on the same grid NPCs and projectiles have their positions on.